src/components/TimeseriesDataExport/TimeseriesDataExport.tsx

```diff
@@ -103,7 +86,14 @@ const TimeseriesDataExportFC: FC<TimeseriesDataExportFormProps> = (
   const [limitHit, setLimitHit] = useState(false);
   const [loading, setLoading] = useState(false);
   const [series, setSeries] = useState<GetTimeSeriesMetadataDTO[]>([]);
-  const lang = { ...defaultStrings, ...strings };
+  const lang = useMemo(
+    () => ({
+      ...defaultStrings,
+      ...(isFunction(strings) ? strings(defaultStrings) : strings),
+    }),
+    [strings]
```

Comment:
`useMemo` here makes sense only if you'll pass `string` function from some helper or something like this. If you'll pass it like an arrow function or define somewhere inside FC – optimization will not work. But I like this for sure 😄

Reply:
`useCallback` to the rescue 😉